src/wallet/wallet.cpp

```diff
@@ -2739,8 +2747,16 @@ bool CWallet::CreateTransaction(const std::vector<CRecipient>& vecSend, CWalletT
                     return false;
                 }
 
-                LearnRelatedScripts(vchPubKey, g_change_type);
-                scriptChange = GetScriptForDestination(GetDestinationForKey(vchPubKey, g_change_type));
+                // If -changetype is not specified and g_address_type is not legacy,
+                // and any destination is (P2SH-)P2WPKH, use P2WPKH for the change
+                // output.
+                OutputType change_type = g_change_type;
+                if (!gArgs.IsArgSet("-changetype") && g_address_type != OUTPUT_TYPE_LEGACY && any_destination_native_segwit) {
```

Comment:
Suggestion: test g_address_type against OUTPUT_TYPE_DEFAULT here, instead of touching `gArgs` here. 

If that is not possible, please move the `gArgs.IsArgSet("-changetype")` to `wallet/init.cpp` and set a global boolean `changetype_is_default` or such - don't spread the handling for one option over two separate places.

Reply:
`OUTPUT_TYPE_DEFAULT` is used for the help message and is defined as `OUTPUT_TYPE_P2SH_SEGWIT`. I'll try (something along the lines of) @luke-jr's 5efd54745.